pkg/components/prometheus-operator/component.go

```diff
@@ -115,16 +122,26 @@ func newComponent() *component {
 				"tier":    "control-plane",
 			},
 		},
+		Grafana: &Grafana{},
```

Comment:
Do we need it here?

Reply:
Yes, we do need it. Otherwise you will see some nil pointer exceptions.

Reply:
I think those can be avoided by checking them then? It doesn't make much sense to me to put it here.

Reply:
Tests are also passing if I remove it.

Reply:
Ah, I'm able to trigger it when I modified the tests, hmm.

Reply:
Maybe we can make Grafana block not a pointer?

Reply:
What's wrong with it? What do you think would be potential implications if we don't make it a pointer?

Reply:
I'm not sure about the possible implications, but if it's a pointer, it means we should handle the cases when it's `nil`. Also, if defined here, if provides no "default" values, as we usually use `newComponent` for, so I'd remove it.

Please have a look at the patch I posted. It adds the handling and test for it.